src/Foundatio.Redis/Cache/RedisCacheClientOptions.cs

```diff
@@ -16,6 +16,7 @@ public class RedisCacheClientOptions : SharedOptions
     /// </summary>
     public CommandFlags ReadMode { get; set; } = CommandFlags.None;
 
+    public int? DbId { get; set; }
```

Comment:
I know the stack exchange client takes an int, I think we should keep this non nullable and default to -1. Part of my wonders if We should almost convert to a byte because stand alone kind of defaults to max of 16 dbs but on some systems there is no limit..., but then that would be invalid for what the SE client takes for the default.

The one edge case to all of this is we don't have any configuration to detect cluster mode and in cluster mode the database has to be set to 0 (or -1 I'm assuming).

With all of this said I think we should just do

```suggestion
    public int DbId { get; set; } = -1;
```

Reply:
Thanks @TeddyAlbina!

I don't agree with making this non-nullable. I want to know if the intent was to use a specific Db or not.

I think the name of this should be `Database` not `DbId`.

Reply:
For cluster mode detection i can add an INFO command to detect it. So if Database is not null and not equals 0 we first send the INFO command, if we are in cluster mode we throw an error indicating configuration doesn't make sens.